### ISSUE TITLE
Maintain list of EC2 instance types and AWS regions

### ISSUE THREAD

**Issue by pdehaan** [loads/loads-web#60]
Per the de-facto standard (and now infamous) [pushgo.json](https://github.com/loads/loads-broker/blob/master/pushgo.json) file, it seems like we want to spit out a bunch of EC2 instance types (`instance_type`) and AWS regions (`instance_region`).

I can't find any Node modules which give us this info, but I did find a few AWS pages which I scraped and dutifully converted to JSON which we can use (not sure if it's worth publishing these to npm for giggles, or if it's too much to maintain).

Anyways, here's the gist of it:
```js
// http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html
exports.awsRegions = {
  "ap-northeast-1": "Asia Pacific (Tokyo)",
  "ap-southeast-1": "Asia Pacific (Singapore)",
  "ap-southeast-2": "Asia Pacific (Sydney)",
  "eu-central-1": "EU (Frankfurt)",
  "eu-west-1": "EU (Ireland)",
  "sa-east-1": "South America (Sao Paulo)",
  "us-east-1": "US East (N. Virginia)",
  "us-west-1": "US West (N. California)",
  "us-west-2": "US West (Oregon)"
};

// http://aws.amazon.com/ec2/instance-types/
exports.awsTypes = [
  // General Purpose (T2)
  "t2.micro",
  "t2.small",
  "t2.medium",

  // General Purpose (M3)
  "m3.medium",
  "m3.large",
  "m3.xlarge",
  "m3.2xlarge",

  // Compute Optimized (C4)
  "c4.large",
  "c4.xlarge",
  "c4.2xlarge",
  "c4.4xlarge",
  "c4.8xlarge",

  // Compute Optimized (C3)
  "c3.large",
  "c3.xlarge",
  "c3.2xlarge",
  "c3.4xlarge",
  "c3.8xlarge",

  // Memory Optimized (R3)
  "r3.large",
  "r3.xlarge",
  "r3.2xlarge",
  "r3.4xlarge",
  "r3.8xlarge",

  // GPU (G2)
  "g2.2xlarge",

  // Storage Optimized (I2)
  "i2.xlarge",
  "i2.2xlarge",
  "i2.4xlarge",
  "i2.8xlarge",

  // Storage Optimized (HS1)
  "hs1.8xlarge"
];
```

It's a bit ugly, but the `key:value` structure of `exports.awsRegions` allows us to have a nicely formatted `<select>` field, or just grab the short instance names using `Object.keys(awsRegions)` (which would be very useful for validating the Joi schema if we say that the `instance_region` must be one of the supported keys.
Likewise with the simple array of `awsTypes` (pushgo.json's `instance_type` field). We could further filter those by prefix if we only care about and/or support certain instance types. For example, `t2.*` and `m3.*` and `c4.*` and `c3.*`. I didn't try and do any needlessly clever nesting here so it should be easier to manipulate the data.

Anyways, my current thought was we either publish that silly JavaScript as a super-simple npm module and use `require()` to get the data, and maintain it via pull requests in GitHub (as per usual), or we just have them as static files in our repo. I figure it could be useful to store them in the **convict** config, if I can sort that out and get it landed, and then maybe have some _/api/aws_ Hapi server endpoint which returns the data in a JSON blob, so we can cleverly load it via the Angular front end. Or maybe just dump them in a ./lib/aws.js and use basic exports.

If I get some time on Friday or on the weekend I can probably prototype that out; it should be fairly straight forward.

**Comment by pdehaan** [comment]
published to ye olde npm registry: https://www.npmjs.com/package/awstypes

Even added some basic APIs for displaying AWS regions by key/value or just key, plus an API to get EC2 instance types and optionally filter them (like, if you only want "m3.\*", "c4.\*", and "c3.\*" types instead of everything).

Feel free to submit PRs to https://github.com/pdehaan/awstypes if you're so inclined.

**Comment by pdehaan** [comment]
I still need to convert **loads-web** to use the new+stupid [**awstypes**](https://github.com/pdehaan/awstypes) npm module and possibly add to a convict config and Hapi endpoint so we can reused the AWS info between the server and client code.

**Comment by pdehaan** [comment]
Not sure if we want all these instance types for **loads-web**, or if we only want to display `t2.*`, `m3.*`, `c4.*`, and `c3.*`:

```json
{
  "instanceTypes": [
    "t2.micro",
    "t2.small",
    "t2.medium",
    "m3.medium",
    "m3.large",
    "m3.xlarge",
    "m3.2xlarge",
    "c4.large",
    "c4.xlarge",
    "c4.2xlarge",
    "c4.4xlarge",
    "c4.8xlarge",
    "c3.large",
    "c3.xlarge",
    "c3.2xlarge",
    "c3.4xlarge",
    "c3.8xlarge",
    "r3.large",
    "r3.xlarge",
    "r3.2xlarge",
    "r3.4xlarge",
    "r3.8xlarge",
    "g2.2xlarge",
    "i2.xlarge",
    "i2.2xlarge",
    "i2.4xlarge",
    "i2.8xlarge",
    "hs1.8xlarge"
  ],
  "regions": {
    "ap-northeast-1": "Asia Pacific (Tokyo)",
    "ap-southeast-1": "Asia Pacific (Singapore)",
    "ap-southeast-2": "Asia Pacific (Sydney)",
    "eu-central-1": "EU (Frankfurt)",
    "eu-west-1": "EU (Ireland)",
    "sa-east-1": "South America (Sao Paulo)",
    "us-east-1": "US East (N. Virginia)",
    "us-west-1": "US West (N. California)",
    "us-west-2": "US West (Oregon)"
  }
}
```

**Comment by pdehaan** [comment]
Fixed in #55
